src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const newState = { ...state };
+
+  for (const obj of actions) {
```

Comment:
try to use more meaningful naming so that you can understand what is meant `obj.extraData` (action.extraData) or `obj.type` (action.type). 
```suggestion
  for (const action of actions) {
```